File: app/src/main/java/mydebts/android/app/feature/events/EventsAdapter.java
package mydebts.android.app.feature.events;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import mydebts.android.app.R;
import mydebts.android.app.data.model.Event;

class EventsAdapter extends RecyclerView.Adapter<EventsAdapter.EventViewHolder> {
    private List<Event> events;
    private OnEventClickListener listener;

    @Override
    public EventViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        EventViewHolder viewHolder = EventViewHolder.create(parent);

        viewHolder.itemView.setOnClickListener(v -> {
            int position = viewHolder.getAdapterPosition();
            if (listener != null && position >= 0) {
                listener.onEventClick(events.get(position));
            }
        });

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(EventViewHolder holder, int position) {
        holder.date.setText(events.get(position).getDate().toString());
    }

    @Override
    public int getItemCount() {
        return events == null ? 0 : events.size();
    }

    void setEvents(List<Event> events) {
        this.events = events;
        notifyDataSetChanged();
    }

    void setOnEventClickListener(OnEventClickListener listener) {
        this.listener = listener;
    }

    static class EventViewHolder extends RecyclerView.ViewHolder {
        final TextView date;

        EventViewHolder(View itemView) {
            super(itemView);
            date = (TextView)itemView.findViewById(R.id.date);
        }

        static EventViewHolder create(ViewGroup parent) {
            return new EventViewHolder(LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.item_event, parent, false));
        }
    }

    interface OnEventClickListener {
        void onEventClick(Event event);
    }
}

File: app/src/main/java/mydebts/android/app/feature/events/EventsAdapter.kt
package mydebts.android.app.feature.events

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import mydebts.android.app.R
import mydebts.android.app.data.model.Event
import mydebts.android.app.extention.toEventDateString

internal class EventsAdapter(
        private val events: List<Event>,
        private val selections: List<Int>) : RecyclerView.Adapter<EventsAdapter.EventViewHolder>() {

    private var onItemClickListener: ((Int) -> Unit)? = null
    private var onItemLongClickListener: ((Int) -> Boolean)? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EventViewHolder {
        val viewHolder = EventViewHolder.create(parent)
        viewHolder.itemView.setOnClickListener { onItemClickListener?.invoke(viewHolder.adapterPosition) }
        viewHolder.itemView.setOnLongClickListener { onItemLongClickListener?.invoke(viewHolder.adapterPosition) ?: false }
        return viewHolder
    }

    override fun onBindViewHolder(holder: EventViewHolder, position: Int) {
        holder.date.text = events[position].date?.toEventDateString()
        holder.itemView.isSelected = selections.contains(position)
    }

    override fun getItemCount(): Int = events.size

    fun setOnItemClickListener(listener: ((Int) -> Unit)) {
        onItemClickListener = listener
    }

    fun setOnItemLongClickListener(listener: ((Int) -> Boolean)) {
        onItemLongClickListener = listener
    }

    internal class EventViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var date: TextView = itemView.findViewById(R.id.text1) as TextView

        companion object {

            fun create(parent: ViewGroup): EventViewHolder {
                return EventViewHolder(LayoutInflater.from(parent.context)
                        .inflate(R.layout.item_one_line, parent, false))
            }
        }
    }
}
